server/channels/app/team.go

```diff
@@ -1503,7 +1503,19 @@ func (a *App) prepareInviteGuestsToChannels(teamID string, guestsInvite *model.G
 		if channel.TeamId != teamID {
 			return nil, nil, nil, model.NewAppError("prepareInviteGuestsToChannels", "api.team.invite_guests.channel_in_invalid_team.app_error", nil, "", http.StatusBadRequest)
 		}
+
+		// Check if the channel has access control policy
+		ctx := request.EmptyContext(a.Log())
+		isControlled, err := a.ChannelAccessControlled(ctx, channel.Id)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		if isControlled {
+			return nil, nil, nil, model.NewAppError("prepareInviteGuestsToChannels", "api.team.invite_guests.abac_policy_enforced_channel.app_error", nil, "", http.StatusBadRequest)
+		}
```

Comment:
I think we can simply use
```suggestion
		if channel.PolicyEnforced {
			return nil, nil, nil, model.NewAppError("prepareInviteGuestsToChannels", "api.team.invite_guests.abac_policy_enforced_channel.app_error", nil, "", http.StatusBadRequest)
		}
```